src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -1,351 +1,73 @@
-                    if (method.MethodKind == MethodKind.ExplicitInterfaceImplementation &&
-                        method.ExplicitInterfaceImplementations.Any(m => !_symbolFilter.Include(m.ContainingSymbol) ||
-                        // if explicit interface implementation method has inaccessible type argument
-                        m.ContainingType.HasInaccessibleTypeArgument(_symbolFilter)))
-                    {
-                        continue;
-                    }
-
-                    // Filter out default constructors since these will be added automatically
-                    if (method.IsImplicitDefaultConstructor(_symbolFilter))
-                    {
-                        continue;
-                    }
-                }
-
-                // If the property is derived from an interface that was filter out, we must filtered out it either.
-                if (member is IPropertySymbol property && !property.ExplicitInterfaceImplementations.IsEmpty &&
-                    property.ExplicitInterfaceImplementations.Any(m => !_symbolFilter.Include(m.ContainingSymbol)))
-                {
-                    continue;
-                }
-
-                SyntaxNode memberDeclaration = _syntaxGenerator
-                    .DeclarationExt(member, _symbolFilter)
-                    .AddMemberAttributes(_syntaxGenerator, member, _attributeDataSymbolFilter);
-
-                if (member is INamedTypeSymbol nestedTypeSymbol)
-                {
-                    memberDeclaration = Visit(memberDeclaration, nestedTypeSymbol);
-                }
-
-                if (HidesBaseMember(member))
-                {
-                    DeclarationModifiers mods = _syntaxGenerator.GetModifiers(memberDeclaration);
-                    memberDeclaration = _syntaxGenerator.WithModifiers(memberDeclaration, mods.WithIsNew(isNew: true));
-                }
-
-                try
-                {
-                    namedTypeNode = _syntaxGenerator.AddMembers(namedTypeNode, memberDeclaration);
-                }
-                catch (InvalidOperationException e)
-                {
-                    // re-throw the InvalidOperationException with the symbol that caused it.
-                    throw new InvalidOperationException(string.Format(Resources.AddMemberThrowsException,
-                        member.ToDisplayString(),
-                        namedTypeNode,
-                        e.Message));
-                }
-            }
-
-            return namedTypeNode;
+            _textWriter.Write(GetFormattedHeader(_header));
+            Document document = _docGenerator.GetDocumentForAssembly(assemblySymbol);
+            _docGenerator.GetFormattedRootNodeForDocument(document).WriteTo(_textWriter);
         }
 
-        private SyntaxNode GenerateAssemblyAttributes(IAssemblySymbol assembly, SyntaxNode compilationUnit)
+        private static string GetFormattedHeader(string? customHeader)
```

Comment:
This used to be in `GenApiApp.cs` and I moved it here as only CSharpFileBuilder consumers need it (basically just GenApiApp), but not ApiDiff.